### PR TITLE
removed Tilt specific and extraneous stuff

### PR DESCRIPTION
**Pull request by ironcamel**
I recently had a conversation with someone in #tilt channel in IRC and they mentioned our style guide. That's when I realized our style guide is public. This PR removes references to our Tilt specific modules which would not be relevant to the public. And btw, the Tilt specific stuff is not even up to date with the Tilt module renames. I also remove some other things that I feel are not very valuable. I would prefer to keep our style guide on the terse side, and not reproduce existing documentation from Moose, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,16 +152,6 @@ __PACKAGE__->meta->make_immutable;
 1;
 ```
 
-### Never override new
```
Comment (ironcamel) [inline README.md:204]: This is a common sense thing that is already well documented in Moose. I don't think it is very valuable for us to reproduce in our style guide.

File: README.md [hunk 2]
```diff
@@ -152,18 +115,6 @@ my $vars = {
 };
 ```
 
-### Prefer vertically aligned assignments
```
Comment (ironcamel) [inline README.md:155]: Assignments are not always vertically aligned in our code base. I don't think this should be a hard rule. IMO, I think it should be an "at your discretion" thing.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,6 @@
 
 * [General Perl](#general-perl) – Best practices for [Perl](https://www.perl.org), our API's main language
 * [Moose](#moose) – Best practices for [Moose](https://metacpan.org/pod/Moose), our object system
-* [DBIC](#dbic) – Best practices for [DBIx::Class](https://metacpan.org/pod/DBIx::Class), our ORM library
```
Comment (godsflaw) [inline README.md:7]: why did you remove this bit?

**Comment by godsflaw** [comment]
+1, agreed that this is too specific for public consumption.

File: README.md (hunk 3 above)
Comment (ironcamel) [inline README.md:7]: I removed this because the DBIC section is empty.

File: README.md (hunk 2 above)
Comment (tildedave) [inline README.md:155]: +1 on this - 'at your discretion' is an important thing to allow in a style guide
